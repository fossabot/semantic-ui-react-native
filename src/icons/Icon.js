import PropTypes from 'prop-types';
import React from 'react';
import AntDesign from 'react-native-vector-icons/AntDesign';
import Entypo from 'react-native-vector-icons/Entypo';
import EvilIcons from 'react-native-vector-icons/EvilIcons';
import Feather from 'react-native-vector-icons/Feather';
import FontAwesome from 'react-native-vector-icons/FontAwesome';
import FontAwesome5 from 'react-native-vector-icons/FontAwesome5';
import Fontisto from 'react-native-vector-icons/Fontisto';
import Foundation from 'react-native-vector-icons/Foundation';
import Ionicons from 'react-native-vector-icons/Ionicons';
import MaterialCommunityIcons from 'react-native-vector-icons/MaterialCommunityIcons';
import MaterialIcons from 'react-native-vector-icons/MaterialIcons';
import Octicons from 'react-native-vector-icons/Octicons';
import SimpleLineIcons from 'react-native-vector-icons/SimpleLineIcons';
import Zocial from 'react-native-vector-icons/Zocial';

class Icon extends React.Component {
  state = {
    Icon: null,
  };

  componentDidMount() {
    this.setIcon(this.props.type);
  }

  setIcon(type) {
    let iconType;
    switch (type) {
      case 'AntDesign':
        iconType = AntDesign;
        break;
      case 'Entypo':
        iconType = Entypo;
        break;
      case 'EvilIcons':
        iconType = EvilIcons;
        break;
      case 'Feather':
        iconType = Feather;
        break;
      case 'FontAwesome':
        iconType = FontAwesome;
        break;
      case 'FontAwesome5':
        iconType = FontAwesome5;
        break;
      case 'Fontisto':
        iconType = Fontisto;
        break;
      case 'Foundation':
        iconType = Foundation;
        break;
      case 'Ionicons':
        iconType = Ionicons;
        break;
      case 'MaterialCommunityIcons':
        iconType = MaterialCommunityIcons;
        break;
      case 'MaterialIcons':
        iconType = MaterialIcons;
        break;
      case 'Octicons':
        iconType = Octicons;
        break;
      case 'SimpleLineIcons':
        iconType = SimpleLineIcons;
        break;
      case 'Zocial':
        iconType = Zocial;
        break;
      default:
        iconType = Ionicons;
    }
    this.setState({
      Icon: iconType,
    });
  }

  render() {
    return this.state.Icon ? (
      <this.state.Icon
        name={this.props.name}
        color={this.props.color}
        size={this.props.size}
        style={this.props.style}
      />
    ) : null;
  }
}

Icon.propTypes = {
  name: PropTypes.string,
  type: PropTypes.oneOf([
    'AntDesign',
    'Entypo',
    'EvilIcons',
    'Feather',
    'FontAwesome',
    'FontAwesome5',
    'Fontisto',
    'Foundation',
    'Ionicons',
    'MaterialCommunityIcons',
    'MaterialIcons',
    'Octicons',
    'SimpleLineIcons',
    'Zocial',
  ]),
  size: PropTypes.number,
  color: PropTypes.string,
  style: PropTypes.object,
};

export default Icon;
