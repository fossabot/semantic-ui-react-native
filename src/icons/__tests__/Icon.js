import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import Icon from '../Icon';

describe('Icon Component', () => {
	it('should render without issues', () => {
		const component = shallow(<Icon />);

		expect(component.length).toBe(1);
		expect(toJson(component)).toMatchSnapshot();
	});

	it('should render with name', () => {
		const component = shallow(<Icon name="add-circle" />);

		expect(component.length).toBe(1);
		expect(toJson(component)).toMatchSnapshot();
	});

	it('should render with name, type', () => {
		const component = shallow(<Icon name="info" type="AntDesign" />);

		expect(component.length).toBe(1);
		expect(toJson(component)).toMatchSnapshot();
	});

	it('should render with name, type, color', () => {
		const component = shallow(
			<Icon name="500px" type="Entypo" color="#181896" />
		);

		expect(component.length).toBe(1);
		expect(toJson(component)).toMatchSnapshot();
	});

	it('should render with name, type, color, size', () => {
		const component = shallow(
			<Icon name="archive" type="EvilIcons" color="#181896" size={50} />
		);

		expect(component.length).toBe(1);
		expect(toJson(component)).toMatchSnapshot();
	});

	it('should render with name, type, color, size, style', () => {
		const component = shallow(
			<Icon
				name="activity"
				type="Feather"
				color="#181896"
				size={50}
				style={{ position: 'absolute', top: 5, left: 5 }}
			/>
		);

		expect(component.length).toBe(1);
		expect(toJson(component)).toMatchSnapshot();
	});

	it('should render with name, type, color, size, style', () => {
		const component = shallow(
			<Icon
				name="glass"
				type="FontAwesome"
				color="#181896"
				size={50}
				style={{ position: 'absolute', top: 5, left: 5 }}
			/>
		);

		expect(component.length).toBe(1);
		expect(toJson(component)).toMatchSnapshot();
	});

	it('should render with name, type, color, size, style', () => {
		const component = shallow(
			<Icon
				name="ad"
				type="FontAwesome5"
				color="#181896"
				size={50}
				style={{ position: 'absolute', top: 5, left: 5 }}
			/>
		);

		expect(component.length).toBe(1);
		expect(toJson(component)).toMatchSnapshot();
	});

	it('should render with name, type, color, size, style', () => {
		const component = shallow(
			<Icon
				name="aws"
				type="Fontisto"
				color="#181896"
				size={50}
				style={{ position: 'absolute', top: 5, left: 5 }}
			/>
		);

		expect(component.length).toBe(1);
		expect(toJson(component)).toMatchSnapshot();
	});

	it('should render with name, type, color, size, style', () => {
		const component = shallow(
			<Icon
				name="alert"
				type="Foundation"
				color="#181896"
				size={50}
				style={{ position: 'absolute', top: 5, left: 5 }}
			/>
		);

		expect(component.length).toBe(1);
		expect(toJson(component)).toMatchSnapshot();
	});

	it('should render with name, type, color, size, style', () => {
		const component = shallow(
			<Icon
				name="add"
				type="Ionicons"
				color="#181896"
				size={50}
				style={{ position: 'absolute', top: 5, left: 5 }}
			/>
		);

		expect(component.length).toBe(1);
		expect(toJson(component)).toMatchSnapshot();
	});

	it('should render with name, type, color, size, style', () => {
		const component = shallow(
			<Icon
				name="add"
				type="Ionicons"
				color="#181896"
				size={50}
				style={{ position: 'absolute', top: 5, left: 5 }}
			/>
		);

		expect(component.length).toBe(1);
		expect(toJson(component)).toMatchSnapshot();
	});

	it('should render with name, type, color, size, style', () => {
		const component = shallow(
			<Icon
				name="alert"
				type="MaterialCommunityIcons"
				color="#181896"
				size={50}
				style={{ position: 'absolute', top: 5, left: 5 }}
			/>
		);

		expect(component.length).toBe(1);
		expect(toJson(component)).toMatchSnapshot();
	});

	it('should render with name, type, color, size, style', () => {
		const component = shallow(
			<Icon
				name="360"
				type="MaterialIcons"
				color="#181896"
				size={50}
				style={{ position: 'absolute', top: 5, left: 5 }}
			/>
		);

		expect(component.length).toBe(1);
		expect(toJson(component)).toMatchSnapshot();
	});

	it('should render with name, type, color, size, style', () => {
		const component = shallow(
			<Icon
				name="alert"
				type="Octicons"
				color="#181896"
				size={50}
				style={{ position: 'absolute', top: 5, left: 5 }}
			/>
		);

		expect(component.length).toBe(1);
		expect(toJson(component)).toMatchSnapshot();
	});

	it('should render with name, type, color, size, style', () => {
		const component = shallow(
			<Icon
				name="user"
				type="SimpleLineIcons"
				color="#181896"
				size={50}
				style={{ position: 'absolute', top: 5, left: 5 }}
			/>
		);

		expect(component.length).toBe(1);
		expect(toJson(component)).toMatchSnapshot();
	});

	it('should render with name, type, color, size, style', () => {
		const component = shallow(
			<Icon
				name="acrobat"
				type="Zocial"
				color="#181896"
				size={50}
				style={{ position: 'absolute', top: 5, left: 5 }}
			/>
		);

		expect(component.length).toBe(1);
		expect(toJson(component)).toMatchSnapshot();
	});
});
