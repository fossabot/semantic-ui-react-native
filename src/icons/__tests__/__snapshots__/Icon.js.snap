// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon Component should render with name 1`] = `
<Icon
  allowFontScaling={false}
  name="add-circle"
  size={12}
/>
`;

exports[`Icon Component should render with name, type 1`] = `
<Icon
  allowFontScaling={false}
  name="info"
  size={12}
/>
`;

exports[`Icon Component should render with name, type, color 1`] = `
<Icon
  allowFontScaling={false}
  color="#181896"
  name="500px"
  size={12}
/>
`;

exports[`Icon Component should render with name, type, color, size 1`] = `
<Icon
  allowFontScaling={false}
  color="#181896"
  name="archive"
  size={50}
/>
`;

exports[`Icon Component should render with name, type, color, size, style 1`] = `
<Icon
  allowFontScaling={false}
  color="#181896"
  name="activity"
  size={50}
  style={
    Object {
      "left": 5,
      "position": "absolute",
      "top": 5,
    }
  }
/>
`;

exports[`Icon Component should render with name, type, color, size, style 2`] = `
<Icon
  allowFontScaling={false}
  color="#181896"
  name="glass"
  size={50}
  style={
    Object {
      "left": 5,
      "position": "absolute",
      "top": 5,
    }
  }
/>
`;

exports[`Icon Component should render with name, type, color, size, style 3`] = `
<IconClass
  brand={false}
  color="#181896"
  light={false}
  name="ad"
  regular={false}
  size={50}
  solid={false}
  style={
    Object {
      "left": 5,
      "position": "absolute",
      "top": 5,
    }
  }
/>
`;

exports[`Icon Component should render with name, type, color, size, style 4`] = `
<Icon
  allowFontScaling={false}
  color="#181896"
  name="aws"
  size={50}
  style={
    Object {
      "left": 5,
      "position": "absolute",
      "top": 5,
    }
  }
/>
`;

exports[`Icon Component should render with name, type, color, size, style 5`] = `
<Icon
  allowFontScaling={false}
  color="#181896"
  name="alert"
  size={50}
  style={
    Object {
      "left": 5,
      "position": "absolute",
      "top": 5,
    }
  }
/>
`;

exports[`Icon Component should render with name, type, color, size, style 6`] = `
<Icon
  allowFontScaling={false}
  color="#181896"
  name="add"
  size={50}
  style={
    Object {
      "left": 5,
      "position": "absolute",
      "top": 5,
    }
  }
/>
`;

exports[`Icon Component should render with name, type, color, size, style 7`] = `
<Icon
  allowFontScaling={false}
  color="#181896"
  name="add"
  size={50}
  style={
    Object {
      "left": 5,
      "position": "absolute",
      "top": 5,
    }
  }
/>
`;

exports[`Icon Component should render with name, type, color, size, style 8`] = `
<Icon
  allowFontScaling={false}
  color="#181896"
  name="alert"
  size={50}
  style={
    Object {
      "left": 5,
      "position": "absolute",
      "top": 5,
    }
  }
/>
`;

exports[`Icon Component should render with name, type, color, size, style 9`] = `
<Icon
  allowFontScaling={false}
  color="#181896"
  name="360"
  size={50}
  style={
    Object {
      "left": 5,
      "position": "absolute",
      "top": 5,
    }
  }
/>
`;

exports[`Icon Component should render with name, type, color, size, style 10`] = `
<Icon
  allowFontScaling={false}
  color="#181896"
  name="alert"
  size={50}
  style={
    Object {
      "left": 5,
      "position": "absolute",
      "top": 5,
    }
  }
/>
`;

exports[`Icon Component should render with name, type, color, size, style 11`] = `
<Icon
  allowFontScaling={false}
  color="#181896"
  name="user"
  size={50}
  style={
    Object {
      "left": 5,
      "position": "absolute",
      "top": 5,
    }
  }
/>
`;

exports[`Icon Component should render with name, type, color, size, style 12`] = `
<Icon
  allowFontScaling={false}
  color="#181896"
  name="acrobat"
  size={50}
  style={
    Object {
      "left": 5,
      "position": "absolute",
      "top": 5,
    }
  }
/>
`;

exports[`Icon Component should render without issues 1`] = `
<Icon
  allowFontScaling={false}
  size={12}
/>
`;
